### PR TITLE
Added CRUD implementation for Deal API

### PR DESCRIPTION
**Pull request by JoachimS**
This includes unit and functional tests.

Additionally this Deserializer now accepts the dealId as a root value for Id.

### CODE REVIEW

File: src/hubspot-client.csproj [hunk 1]
```diff
@@ -21,6 +21,7 @@
     <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
     <PackageReference Include="RapidCore" Version="0.14.0" />
     <PackageReference Include="System.Reflection.TypeExtensions" Version="4.3.0" />
+    <PackageReference Include="xunit" Version="2.2.0" />
```
Comment (nover) [inline src/hubspot-client.csproj]: Main project shouldn't depend on xunit :)

File: test/functional/Deal/HubSpotDealClientFunctionalTest.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+using Skarp.HubSpotClient.Deal;
+using Skarp.HubSpotClient.Deal.Dto;
+using Skarp.HubSpotClient.Core.Requests;
+using Xunit;
+using Xunit.Abstractions;
+using Skarp.HubSpotClient.FunctionalTests.Mocks.Deal;
+
+namespace Skarp.HubSpotClient.FunctionalTests.Deal
+{
+    public class HubSpotDealClientFunctionalTest : FunctionalTestBase<HubSpotDealClient>
```
Comment (nover) [inline test/functional/Deal/HubSpotDealClientFunctionalTest.cs]: nice with a functional test for it all :100:

Comment (JoachimS) [reply test/functional/Deal/HubSpotDealClientFunctionalTest.cs]: Updated the project dependency - never did that addition intentionally 👍

Comment (nover) [reply test/functional/Deal/HubSpotDealClientFunctionalTest.cs]: @JoachimS Great thanks - and I know :) 👍 
Thanks for the PR.

**Pull request merged**